src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4363,6 +4398,9 @@ void os::init(void) {
 
   check_pax();
 
+  // Check the availability of MADV_POPULATE_WRITE.
+  UseMadvPopulateWrite = (::madvise(0, 0, MADV_POPULATE_WRITE) == 0);
+
```

Comment:
What happens if the user sets `UseMadvPopulateWrite` to false when starting the JVM? It should not be used then.

Reply:
It is before Arguments::parse, so I would use FLAG_SET_DEFAULT here.